spec.bs

```diff
@@ -268,13 +290,28 @@ Insert a call to [=wait for import maps=] at the beginning of the following HTML
 <h3 id="integration-register-an-import-map">Registering an import map</h3>
 
 <div algorithm>
-To <dfn>register an import map</dfn> given an [=environment settings object=] |settings object| and an {{HTMLScriptElement}} |element|:
+To <dfn>register an import map</dfn> given an {{HTMLScriptElement}} |element|:
 
 1. If |element|'s [=the script's result=] is null, then [=queue a task=] to [=fire an event=] named `error` at |element|, and return.
+1. Let |import map parse result| be |element|'s [=the script's result=].
+1. Assert: |element|'s <a spec="html">the script's type</a> is "`importmap`".
+1. Assert: |import map parse result| is an [=import map parse result=].
+1. Let |settings object| be |import map parse result|'s [=import map parse result/settings object=].
 1. If |element|'s <a spec="html">node document</a>'s <a spec="html">relevant settings object</a> is not equal to |settings object|, then return.
    <p class="note">This is spec'ed consistently with <a href="https://github.com/whatwg/html/pull/2673">whatwg/html#2673</a>.</p>
    <p class="advisement">Currently we don't fire `error` events in this case. If we change the decision at <a href="https://github.com/whatwg/html/pull/2673">whatwg/html#2673</a> to fire `error` events, then we should change this step accordingly.</p>
-1. Otherwise, [=update an import map|update=] |element|'s [=node document=]'s [=Document/import map=] with |element|'s [=the script's result=].
+1. If |import map parse result|'s [=import map parse result/error to rethrow=] is not null, then:
+  1. <div class="issue">
+       Should we do the following, in order to match the behavior with <a spec="html">execute a script block</a>?
+       1. Increment/decrement <a spec="html">ignore-destructive-writes counter</a> of |element|'s [=node document=].
+       1. <a spec="html">Check if we can run script</a> with |settings object|. If this returns "`do not run`" then return.
+       1. <a spec="html">Prepare to run script</a> given |settings object|.
+       1. <a spec="html">Clean up after running script</a> given |settings object|.
```

Comment:
I am pretty sure the only potentially observable thing here would be the "check if we can run script" step.

I guess that would be observable because it would make inserting import maps into non-fully active documents a no-op, which you could then observe by making the document fully active again? (Which I think can only happen with bfcache.) Wow, what an edge case.

Reply:
I'll remove these for now.